app/src/main/java/co/nimblehq/compose/crypto/ui/theme/AppColors.kt

```diff
@@ -0,0 +1,65 @@
+package co.nimblehq.compose.crypto.ui.theme
+
+import androidx.compose.material.*
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+
+/**
+ * Extend final [Colors] class to provide more custom app colors.
+ */
+data class AppColors(
+    val themeColors: Colors,
+
+    val coinChartLineDrawer: Color = CaribbeanGreen,
+    val coinChartLineShaderGradient: List<Color> = listOf(CaribbeanGreenAlpha30, Transparent),
+    val coinChartTimeIntervalBackground: Color = Transparent,
+    val coinChartTimeIntervalBackgroundSelected: Color = CaribbeanGreen,
+    val coinChartTimeIntervalText: Color = OsloGray,
+    val coinChartTimeIntervalTextSelected: Color = White,
+    val coinInfoAppBarIconTint: Color,
+    val coinInfoSellBuyBackground: Color,
+    val coinInfoSellBuyText: Color = White,
+    val coinItemBackground: Color,
+    val coinNameText: Color,
+
+    val buttonBlueBackground: Color = TiffanyBlue,
+    val buttonRedBackground: Color = FireOpal,
+
+    val portfolioCardBackgroundGradient: List<Color> = listOf(MetallicSeaweed, TiffanyBlue),
+    val portfolioCardShadow: Color = TiffanyBlue,
+    val priceChangeButtonBackground: Color = Water,
+    val priceChangeButtonBackgroundInDetail: Color = WhiteIce,
+    val priceTextNegative: Color = FireOpal,
+    val priceTextPositive: Color = GuppieGreen,
+    val pullRefreshBackground: Color,
+    val pullRefreshContent: Color = CaribbeanGreen,
+
+    val text: Color,
+    val textSectionLink: Color = TiffanyBlue,
+)
+
+internal val LightColorPalette = AppColors(
+    themeColors = lightColors(
+        surface = Guyabano,
+    ),
+    coinInfoAppBarIconTint = Quartz,
+    coinInfoSellBuyBackground = White,
+    coinItemBackground = White,
+    coinNameText = SonicSilver,
+    pullRefreshBackground = White,
+    text = DarkJungleGreen
+)
+
+internal val DarkColorPalette = AppColors(
+    themeColors = darkColors(
+        surface = DarkJungleGreen
+    ),
+    coinInfoAppBarIconTint = White,
+    coinInfoSellBuyBackground = DarkJungleGreen,
+    coinItemBackground = QuartzAlpha20,
+    coinNameText = LightSilver,
+    pullRefreshBackground = DarkJungleGreen,
+    text = White
+)
+
+internal val LocalColors = staticCompositionLocalOf { LightColorPalette }
```

Comment:
@luongvo Following the official [document](https://developer.android.com/jetpack/compose/compositionlocal#creating-apis), I think we don't need to apply the `staticCompositionLocalOf` in here, it will recompose all the `content` if the value changes 🤔 

`Changing the value causes the entirety of the content lambda where the CompositionLocal is provided to be recomposed, instead of just the places where the current value is read in the Composition.`

Reply:
@hoangnguyen92dn I think that's our expectation for theme resources? `LocalColors` keeps the reference to our current AppColors value for each light or dark theme. Switching between light and dark themes will set the new value to `LocalColors` and it should be applied on the whole app UI, right?

There are some references for us here:

https://github.com/android/compose-samples/blob/11ae75b86f15ee5370a87631026697f22e778514/Owl/app/src/main/java/com/example/owl/ui/theme/Elevations.kt#L30

https://github.com/android/compose-samples/blob/11ae75b86f15ee5370a87631026697f22e778514/Owl/app/src/main/java/com/example/owl/ui/theme/Images.kt#L29

https://github.com/android/compose-samples/blob/13a16a0553e43a6165c638749d2cf0058aab6de5/Jetsnack/app/src/main/java/com/example/jetsnack/ui/theme/Theme.kt#L284

https://github.com/MakeItEasyDev/Jetpack-Compose-Support-Different-Screen-Size/blob/e70886f23ec86cf571088e220a7f8471bf8754d2/app/src/main/java/com/jetpack/supportdifferentscreensize/ui/theme/Theme.kt#L32

https://gustav-karlsson.medium.com/extending-the-jetpack-compose-material-theme-with-more-colors-e1b849390d50

https://medium.com/mobile-app-development-publication/android-jetpack-compose-compositionlocal-made-easy-8632b201bfcd

Reply:
@hoangnguyen92dn I think it meets of concept to have `CompositionLocal` for the current implementation. Here's the ref form [Jetpack Compose Playground](https://foso.github.io/Jetpack-Compose-Playground/general/compositionlocal/)

> CompositionLocal is useful when you want to create a dependency in a higher node of the layout tree and use it on lower node without having to pass it down the tree through every child Composable.

Regarding the definition above ☝️ I think it's good as we will use `object AppTheme` to get the values of a theme. Is I understand your purpose correctly @luongvo 🤔

Reply:
@Wadeewee @luongvo I mean we could make use of `compositionLocalOf ` instead 🤔

Reply:
@hoangnguyen92dn Yes, and I explained here https://github.com/nimblehq/jetpack-compose-crypto/pull/66#discussion_r1023459642